packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -148,64 +157,95 @@ export function toVisualTree(tree: VisualSharedTreeNode): VisualChildNode {
 }
 
 /**
- * Concatenrate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
+ * Concatenate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
  */
-function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
+export function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
 	return [...fieldTypes].join(" | ");
 }
 
 /**
- * Returns the allowed fields & types for the object fields (e.g., `foo : string | number, bar: boolean`)
+ * Determines if a field or node is required based on its schema kind.
+ * @param schema - The schema to check, either a SimpleTreeSchema or SimpleFieldSchema
+ * @returns "true" if the field is required (FieldKind.Required), "false" otherwise
+ *
+ * @remarks
+ * This function is used when visualizing schema information to indicate whether
+ * fields in object nodes are required or optional. The return value is a string
+ * rather than a boolean to match the expected format for schema visualization.
  */
-function getObjectAllowedTypes(schema: SimpleObjectNodeSchema): string {
-	const result: string[] = [];
-
-	for (const [fieldKey, treeFieldSimpleSchema] of Object.entries(schema.fields)) {
-		const fieldTypes = treeFieldSimpleSchema.allowedTypes;
-		result.push(`${fieldKey} : ${concatenateTypes(fieldTypes)}`);
-	}
-
-	return `{ ${result.join(", ")} }`;
+export function getIsRequired(schema: SimpleTreeSchema | SimpleFieldSchema): string {
+	return schema.kind === FieldKind.Required ? "true" : "false";
 }
 
 /**
  * Returns the schema & fields of the node.
  */
 async function visualizeVerboseNodeFields(
-	tree: VerboseTreeNode,
-	treeSchema: SimpleTreeSchema,
+	treeFields: VerboseTree[] | Record<string, VerboseTree>,
+	treeDefinitions: ReadonlyMap<string, SimpleNodeSchema>,
+	allowedTypes: Record<string, string>,
+	requirements: Record<string, string> | undefined,
 	visualizeChildData: VisualizeChildData,
 ): Promise<Record<string, VisualSharedTreeNode>> {
-	const treeFields = tree.fields;
-
 	const fields: Record<string | number, VisualSharedTreeNode> = {};
 
 	for (const [fieldKey, childField] of Object.entries(treeFields)) {
-		fields[fieldKey] = await visualizeSharedTreeNodeBySchema(
+		fields[fieldKey] = await visualizeSharedTreeBySchema(
 			childField,
-			treeSchema,
+			treeDefinitions,
+			allowedTypes[fieldKey],
+			requirements === undefined ? requirements : requirements[fieldKey],
 			visualizeChildData,
 		);
 	}
 
 	return fields;
 }
 
+/**
+ * Extracts and stores allowed types & kind from {@link SimpleFieldSchema}.
+ */
+function storeObjectAllowedTypes(schema: SimpleObjectNodeSchema): {
+	allowedTypes: Record<string, string>;
+	requirements: Record<string, string>;
```

Comment:
Were we going to change this from `requirements: required / optional / identifier` to `required: true / false`?

Reply:
You're right. With the removal of the `getRequirement()`, I am simply filtering based on the `Required` & `Optional` kinds. 

Is the name `requirements` misleading?

Reply:
It's a bit less specific, yeah

Reply:
Changed.